docs/devguide/xtypes.rst

```diff
@@ -1315,10 +1331,135 @@ They provides the same operations as any other DataWriter or DataReader, but wit
 See :ref:`xtypes--populating-data-samples-with-dynamicdata` for details on creating DynamicData objects for use with the DynamicDataWriter interface.
 See :ref:`xtypes--interpreting-data-samples-with-dynamicdata` for details on using DynamicData objects obtained from the DynamicDataReader interface.
 
+.. _xtypes-dynamicdata-and-idl-generated-types:
+
+``DynamicData`` and IDL-Generated Types
+=======================================
+
+``DynamicData`` allows writing code that is much more generic than using the IDL-generated types.
+There are still reasons to use the IDL-generated types though.
+They are more convenient to use and are more likely to be used in existing code.
+You can still make use of the flexibility of ``DynamicData`` though by either interchanging with or wrapping IDL-generated types.
+
+Interchanging Data using ``create_sample`` and ``create_dynamic_sample``
+------------------------------------------------------------------------
+
+.. versionadded:: 3.29.0
+
+The XTypes specification defines ``create_sample`` and ``create_dynamic_sample`` on the ``TypeSupport`` for every :term:`topic type` to interchange ``DynamicData`` and IDL-generated types.
+In addition to this, OpenDDS adds ``create_sample_rc`` and ``create_dynamic_sample_rc`` to indicate failure and provide a more consistent interface.
+The standard methods will only warn if there was an error and can't indicate that an error occured.
+
+``create_sample`` and ``create_sample_rc``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``create_sample`` takes a ``DynamicData`` and returns the IDL-generated type.
+One thing to note about this method is the C++ to IDL specification requires that returns like this can be either returned on the stack or the heap depending on if they are :omgspec:`"variable-length" <cpp03:5.11 Mapping for Structured Types>`.
```

Comment:
```suggestion
One thing to note about this method is the IDL-to-C++ specification requires that returns like this can be either returned on the stack or the heap depending on if they are :omgspec:`"variable-length" <cpp03:5.11 Mapping for Structured Types>`.
```

Reply:
Why do I keep doing that?